cobbler/modules/managers/import_signatures.py

```diff
@@ -212,13 +212,11 @@ def run(
                     )
                     if rc == 0:
                         cmd = [
-                            "/usr/bin/wimdir %s 1 | /usr/bin/grep -i '^/Windows/Boot/PXE$'"
-                            % winpe_path
+                            f"/usr/bin/wimdir {winpe_path} 1 | /usr/bin/grep -i '^/Windows/Boot/PXE$'"
```

Comment:
This is another case where the string could be split.

Reply:
Probably not, with no shell, pipes will not work here.

Reply:
Ah, right. Does this even work then?

Reply:
I will refactor this after the weekend. The WIP commit is already pushed.

Reply:
The thing is that I need to determine the output of `wimdir` to find a suitable expression in Python to replace `grep`.

Reply:
@tpw56j Do you maybe have an example output for this? I am having a hard time actually setting this up sadly...

Reply:
@SchoolGuy  wimlib is case sensitive, while Microsoft creates directories in the boot.wim image in random case. In this example, /`w`indows is in lowercase:
```
# /usr/bin/wimdir boot.wim 1 | /usr/bin/grep -i '^/Windows/Boot/PXE$'
/windows/Boot/PXE
```

Reply:
@tpw56j Yes but I want to get rid of the shell pipe and do the grep in Python. Doing this without the pipe has some very important security considerations because the call has to be made with a system shell instead of a Python function. Thus example output of `wimdir` would be superb.

Reply:
@SchoolGuy Okay. We can get a list of boot.wim image files and find the correct directory in it, case insensitive. You can fix one bug at the same time: earlier windir returned an extra line ending character accepted in windows, but now the behavior of windir has changed and this character is no longer displayed when outputting. Therefore, the code now works with an error. To fix it, you need to remove `[0:-1]` when using this line.
````diff
- pxe_path = utils.subprocess_get(cmd, shell=True)[0:-1]
+ pxe_path = utils.subprocess_get(cmd, shell=True)
````
````diff
- config_path = utils.subprocess_get(cmd, shell=True)[0:-1]
+ config_path = utils.subprocess_get(cmd, shell=True)
````
Number of files in boot.wim for windows 10:
```
# /usr/bin/wimdir boot.wim 1 | wc -l
18388
```

Reply:
@tpw56j Okay thanks for the background. Now can you maybe give me a GitHub Gist or pastebin how the output of `wimdir` looks like? WIth that information I can construct the search I desire to replace the grep expression.

Reply:
@SchoolGuy The output of `wimdir` looks the same as the output of `ls -1R /`:
https://gist.github.com/tpw56j/baf68dfb2143c0ffe1a0faf8a839248a